src/main/java/com/ctc/wstx/sax/WstxSAXParser.java

```diff
@@ -621,14 +620,9 @@ public void parse(InputSource input) throws SAXException
             mAttrCollector = mScanner.getAttributeCollector();
             mElemStack = mScanner.getInputElementStack();
             fireEvents();
-        } catch (IOException io) {
-            throwSaxException(io);
-        } catch (XMLStreamException strex) {
-            throwSaxException(strex);
+        } catch (Exception e) {
+            throwSaxException(e);
```

Comment:
@winfriedgerlach Ok, I updated this to basically take all Exceptions, including possible SAXParseException, and handle necessary wrapping and error handler calling.